### PR TITLE
#1568 initial check in of ui admin calls

### PR DESCRIPTION
**Pull request by davidradl**
Signed-off-by: David Radley <david_radley@uk.ibm.com>

### CODE REVIEW

File: open-metadata-implementation/adapters/open-connectors/configuration-store-connectors/pom.xml [hunk 1]
```diff
@@ -37,6 +37,11 @@
 	<dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
+    </dependency>
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>ui-admin-services-api</artifactId>
+            <version>1.1-SNAPSHOT</version>
```
Comment (planetf1) [inline open-metadata-implementation/adapters/open-connectors/configuration-store-connectors/pom.xml]: A version should not be specified here - we only put versions in the top level pom.
Need new issue

File: open-metadata-implementation/user-interfaces/access-services-user-interface/pom.xml [hunk 2]
```diff
@@ -249,8 +249,11 @@
             <artifactId>open-lineage-services-api</artifactId>
             <version>${open-metadata.version}</version>
         </dependency>
-
-
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>ui-admin-services-spring</artifactId>
+            <version>${open-metadata.version}</version>
```
Comment (planetf1) [inline open-metadata-implementation/user-interfaces/access-services-user-interface/pom.xml]: Should not be a version here

File: open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <description>
+        Provides a Java client for the open metadata ui administration services.
+    </description>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>ui-admin-services-api</artifactId>
+            <version>1.1-SNAPSHOT</version>
```
Comment (planetf1) [inline open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-client/pom.xml]: remove version here

File: open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-server/pom.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>ui-admin-services-api</artifactId>
+            <version>${open-metadata.version}</version>
```
Comment (planetf1) [inline open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-server/pom.xml]: remove all references to version in this file on dependencies, managed by top level pom

File: open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-spring/pom.xml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        The ui administration services with Spring module uses Spring to implement a RESTful service
+        for the ui server administration services.
+    </description>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>org.odpi.egeria</groupId>
+            <artifactId>ui-admin-services-api</artifactId>
+            <version>${open-metadata.version}</version>
```
Comment (planetf1) [inline open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-spring/pom.xml]: more removals needed in this file

File: open-metadata-implementation/adapters/open-connectors/configuration-store-connectors/pom.xml (hunk 1 above)
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/configuration-store-connectors/pom.xml]: If not in top level pom, add any new modules to the <dependencyManagement> section there
Ideally <scope> I think should be 'runtime' for all our spring components not references in the code, but discovered during runtime component scanning

**Comment by planetf1** [comment]
If correct though this wasn't merged? (just noticed) so changes may already be in progress @davidradl

**Comment by davidradl** [comment]
I am re doing this code around this - I will see if I can remove the versions. I added them as without them I get Maven errors.

**Comment by davidradl** [comment]
@planetf1  removing the versions gives me these errors:
 
[ERROR]   The project org.odpi.egeria:ui-admin-services-client:1.1-SNAPSHOT (/Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-client/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.odpi.egeria:ui-admin-services-api:jar is missing. @ org.odpi.egeria:ui-admin-services-client:[unknown-version], /Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-client/pom.xml, line 25, column 21
[ERROR]   
[ERROR]   The project org.odpi.egeria:ui-admin-services-server:1.1-SNAPSHOT (/Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-server/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.odpi.egeria:ui-admin-services-api:jar is missing. @ org.odpi.egeria:ui-admin-services-server:[unknown-version], /Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-server/pom.xml, line 28, column 21
[ERROR]   
[ERROR]   The project org.odpi.egeria:ui-admin-services-spring:1.1-SNAPSHOT (/Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-spring/pom.xml) has 2 errors
[ERROR]     'dependencies.dependency.version' for org.odpi.egeria:ui-admin-services-api:jar is missing. @ org.odpi.egeria:ui-admin-services-spring:[unknown-version], /Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-spring/pom.xml, line 26, column 21
[ERROR]     'dependencies.dependency.version' for org.odpi.egeria:ui-admin-services-server:jar is missing. @ org.odpi.egeria:ui-admin-services-spring:[unknown-version], /Users/davidradley/egeria/open-metadata-implementation/user-interfaces/ui-admin-services/ui-admin-services-spring/pom.xml, line 31, column 21
[ERROR]

**Comment by planetf1** [comment]
Have you added the new spring module within the <dependencyManagement> section in the top level pom?

That is a single place where we try to define all versions - it's not declaring a dependency as such there, just that if any submodule uses a dependency, that is the version to use.

**Comment by planetf1** [comment]
See developer-resources/Dependency_Management.md

**Comment by davidradl** [comment]
@planetf1 Perfect - thanks Nigel :-) that was what I had done wrong .

**Pull request closed without merging**